storage/integration_test.go

```diff
@@ -2671,8 +2670,16 @@ func TestIntegration_DeleteObjectInBucketWithRetentionPolicy(t *testing.T) {
 	}
 
 	// Remove the retention period
-	h.mustUpdateBucket(bkt, BucketAttrsToUpdate{RetentionPolicy: &RetentionPolicy{RetentionPeriod: 0}})
-	h.mustDeleteObject(oh)
+	h.mustUpdateBucket(bkt, BucketAttrsToUpdate{RetentionPolicy: &RetentionPolicy{}})
+	// Deleting with retry, as bucket metadata changes
+	// can take some time to propagate.
+	err := runWithRetry(ctx, func() error {
+		time.Sleep(time.Second * 10)
+		return oh.Delete(context.Background())
+	})
+	if err != nil {
+		h.t.Fatalf("%s: object delete: %v", loc(), err)
+	}
```

Comment:
Just adding a retry doesn't give any effect. I've seen in samples, that it's recommended to use a 10 sec pause after bucket metadata update before proceed. Added it, and now it's working fine

Reply:
Did you try to use [this retry function ](https://github.com/googleapis/google-cloud-go/blob/a00f5d931748825e09671257b3c52e90854c2039/storage/integration_test.go#L3596) with 11 second-timeout and it didn't give any effect? Better to test it.
Or you just tried to use runWithRetry without Sleep function and it didn't work?